modules/stochastic_tools/src/utils/GaussianProcess.C

```diff
@@ -67,35 +68,39 @@ GaussianProcess::linkCovarianceFunction(CovarianceFunctionBase * covariance_func
 }
 
 void
-GaussianProcess::setupCovarianceMatrix(const RealEigenMatrix & training_params,
-                                       const RealEigenMatrix & training_data,
+GaussianProcess::setupCovarianceMatrix(const torch::Tensor & training_params,
+                                       const torch::Tensor & training_data,
                                        const GPOptimizerOptions & opts)
 {
-  const bool batch_decision = opts.batch_size > 0 && (opts.batch_size <= training_params.rows());
-  _batch_size = batch_decision ? opts.batch_size : training_params.rows();
-  _K.resize(_num_outputs * _batch_size, _num_outputs * _batch_size);
+  const bool batch_decision =
+      opts.batch_size > 0 && (opts.batch_size <= training_params.sizes()[0]);
+  _batch_size = batch_decision ? opts.batch_size : training_params.sizes()[0];
+  _K = torch::zeros({_num_outputs * _batch_size, _num_outputs * _batch_size}).to(at::kDouble);
 
   if (_tuning_data.size())
     tuneHyperParamsAdam(training_params, training_data, opts);
 
-  _K.resize(training_params.rows() * training_data.cols(),
-            training_params.rows() * training_data.cols());
+  _K = torch::resize(_K,
+                     {training_params.sizes()[0] * training_data.sizes()[1],
+                      training_params.sizes()[0] * training_data.sizes()[1]})
+           .to(at::kDouble);
   _covariance_function->computeCovarianceMatrix(_K, training_params, training_params, true);
+  torch::Tensor training_data_transpose = torch::transpose(training_data, 0, 1);
+  torch::Tensor flattened_tensor = torch::reshape(
+      training_data_transpose, {training_params.sizes()[0] * training_data.sizes()[1], 1});
 
-  RealEigenMatrix flattened_data =
-      training_data.reshaped(training_params.rows() * training_data.cols(), 1);
-
-  // Compute the Cholesky decomposition and inverse action of the covariance matrix
-  setupStoredMatrices(flattened_data);
+  //  Compute the Cholesky decomposition and inverse action of the covariance matrix
+  setupStoredMatrices(flattened_tensor);
 
   _covariance_function->buildHyperParamMap(_hyperparam_map, _hyperparam_vec_map);
 }
 
 void
-GaussianProcess::setupStoredMatrices(const RealEigenMatrix & input)
+GaussianProcess::setupStoredMatrices(const torch::Tensor & input)
 {
-  _K_cho_decomp = _K.llt();
-  _K_results_solve = _K_cho_decomp.solve(input);
+  // llt does the llt decomposition as part of the Cholesky module
+  _K_cho_decomp = torch::linalg_cholesky(_K);
+  _K_results_solve = torch::cholesky_solve(input, _K_cho_decomp);
```

Comment:
I think this will do another cholesky decom of the cholesky (itself) in the background. Would be interesting to see if we could do the two without the double decomposition.